src/runway/tests/handlers/yaml_lint.py

```diff
@@ -49,13 +67,37 @@ def handle(cls, name, args):
                 '.yamllint.yml'
             )
 
-        yl_cmd = "yamllint --config-file=%s" % yamllint_config
-        if dirs_to_scan:
-            yl_cmd += " " + ' '.join(["\"%s\"" % x for x in dirs_to_scan])
-        if files_at_base:
-            yl_cmd += " " + ' '.join(["\"%s\"" % x for x in files_at_base])
+        yamllint_options = ["--config-file=%s" % yamllint_config]
+        yamllint_options.extend(cls.get_yamllint_options(cls,
+                                                         base_dir,
+                                                         not getattr(sys, 'frozen', False)))
+
+        if getattr(sys, 'frozen', False):
+            # running in pyinstaller single-exe, so sys.executable will
+            # be the all-in-one Runway binary
+
+            yamllint_invocation_script = (
+                "import sys;"
+                "from yamllint.cli import run;"
+                "sys.argv = [%s];"
+                "sys.exit(run());" % ','.join(
+                    "'%s'" % i for i in ['yamllint'] + yamllint_options
+                )
+            )
+
+            temp_fd, temp_path = tempfile.mkstemp(prefix='yamllint')
+            os.close(temp_fd)
+            with open(temp_path, 'w') as fileobj:
+                fileobj.write(yamllint_invocation_script)
+
+            yl_cmd = sys.executable + ' run-python ' + temp_path
+        else:
+            # traditional python execution
+            yl_cmd = "yamllint " + ' '.join(yamllint_options)
         with change_dir(base_dir):
             ScriptHandler().handle(
                 'yamllint',
                 {'commands': [yl_cmd]}
             )
+            if getattr(sys, 'frozen', False):
+                os.remove(temp_path)
```

Comment:
This should probably be wrapped in `try: finally` to ensure the temp file is always removed.

```suggestion
        with change_dir(base_dir):
            try:
                ScriptHandler().handle(
                    'yamllint',
                    {'commands': [yl_cmd]}
                )
            finally:
                if getattr(sys, 'frozen', False):
                    os.remove(temp_path)
```

But, would have to double check the functionality to ensure that exceptions raised in `ScriptHandler`, intended to halt execution, still work as intended.

Reply:
I did some basically testing with a finally block and it looks to work as desired with the sys.exits